### DISCUSSION TITLE
Unable to Authenticate other users on the domain

### DISCUSSION THREAD

**Discussion by kevinjoseph61** [danyi1212/django-windowsauth#8] [Q&A] [answered]
I'm able to bind to my LDAP server but no authentication of other users are happening on the domain.
Is there anything I'm missing? I've enabled Windows Authentication but still unable to.. Does anyone have any solutions
The only error I see on the ldap log is ERROR:key 'samaccountname' not present for <OBJ : user

**Comment by danyi1212** [comment, marked as answer]
Hello!

I'm not exactly sure where your problem is. Does the error raised by the Authentication Backend or the LDAP Sync Middleware? (you should see it in the exception's stack trace).

It seems like the LDAP entry comes without the `sAMAccountName` attribute, it could happen either by the client (this module) not asking for it, or the server not having such attribute.

To make sure that your code asks for the `sAMAccountName` attribute, make sure that your `PRELOADED_DEFINITIONS` in your LDAP Settings (`WAUTH_DOMAINS`) for the `user` object class contains it. It should look like that (the default value):

```python
LDAPSettings(
    PRELOAD_DEFINITIONS=(
        ("user", ("sAMAccountName",)),
        "group"
    ),
)
```

To check whether the servers sends the attribute, you can install `django-debug-toolbar` and look for the entries in the [LDAP Panel](https://django-windowsauth.readthedocs.io/en/latest/howto/debug_toolbar.html). You should see the `sAMAccountName` attribute in there.

Can you please send your `WAUTH_DOMAINS` settings? (make sure to not to send sensitive info such as passwords)

**Reply by kevinjoseph61** [reply]
Hi thanks for the quick response
I know that my LDAP server has this attribute sAMAccountName as it the main username
I'm not sure what you want me to do..
```

from windows_auth.ldap import LDAPSettings
WAUTH_DOMAINS = {
    "domain": LDAPSettings(
        SERVER= "ldap.gp.domain.com",
        SEARCH_BASE= "dc=GP,dc=DOMAIN,dc=COM",
        USERNAME= "domain\\user",
        PASSWORD= "password",
        PRELOADED_DEFINITIONS=(
            ("user", ("sAMAccountName",)),
            "group"
        ),
    )
}
```
Is this what you wanted me to do?
```TypeError: __init__() got an unexpected keyword argument 'PRELOADED_DEFINITIONS'```

**Reply by danyi1212** [reply]
Sorry, my mistake. Its called `PRELOAD_DEFINITIONS`, not `PRELOADED_DEFINITIONS` 😅 
https://django-windowsauth.readthedocs.io/en/latest/reference/ldap_settings.html#preload-definitions

Can you please send a screenshot / copy of an entry (from the debug panel)? 
Is `sAMAccountName` listed in the requested attributes? 
And is it returned in the response?

**Reply by kevinjoseph61** [reply]
Hi,
Sorry for the delayed response, I was caught up with some other tasks
I wasn't able to install the toolbar for some reason the URLs weren't displaying correctly but for now I can say that the issue is with the IIS allow users. Since I've enabled only Windows Authentication it was failing but now after enabling Anonymous Authentication I'm able to connect but no LDAP users are being created and credentials are not being checked so I'll have to check with the domain owners now about the credentials
Thanks for your help
